addons/pos_sale/static/src/overrides/models/pos_store.js

```diff
@@ -8,4 +8,14 @@ patch(PosStore.prototype, {
         this.orderManagement = { searchString: "", selectedOrder: null };
         return await super.setup(...args);
     },
+    selectOrderLine(order, line) {
+        super.selectOrderLine(...arguments);
+        if (
+            line &&
+            this.config.down_payment_product_id &&
+            line.product_id.id === line.config.down_payment_product_id.id
```

Comment:
```suggestion
            line.product_id.id === this.config.down_payment_product_id.id
```
Be consistent and use the config linked to the line or to the pos in both places